File: scripts/video-factory/validate_douyin.py
#!/usr/bin/env python3
"""Validate douyin-video.json without launching a browser."""

from __future__ import annotations

import argparse
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parent
sys.path.insert(0, str(ROOT))

from lib import load_json, project_work_dir  # noqa: E402


def validate(work_dir: Path) -> dict:
    douyin_path = work_dir / "douyin-video.json"
    if not douyin_path.exists():
        raise SystemExit(f"Missing {douyin_path}")

    meta = load_json(douyin_path)
    required = ["videoPath", "title", "description", "tags"]
    missing = [k for k in required if not meta.get(k)]
    video = work_dir / str(meta["videoPath"])
    checks = {
        "has_required_fields": not missing,
        "video_exists": video.exists(),
        "title_length_ok": 0 < len(str(meta.get("title", ""))) <= 30,
        "tags_count_ok": 1 <= len(meta.get("tags", [])) <= 5,
    }
    passed = all(checks.values())
    return {"path": str(douyin_path), "checks": checks, "passed": passed, "missing": missing}


def main() -> None:
    parser = argparse.ArgumentParser(description="Validate douyin-video.json")
    parser.add_argument("--id", required=True)
    args = parser.parse_args()
    report = validate(project_work_dir(args.id))
    print(f"Validate {'PASSED' if report['passed'] else 'FAILED'}: {args.id}")
    if not report["passed"]:
        raise SystemExit(1)


if __name__ == "__main__":
    main()

File: scripts/three-kingdoms-english-video/validate-douyin.mjs
#!/usr/bin/env node
/** Validate douyin-video.json without launching browser. */
import fs from "node:fs";
import path from "node:path";

const input = process.argv[2];
if (!input) {
  console.error("Usage: node validate-douyin.mjs <douyin-video.json>");
  process.exit(1);
}

const file = path.resolve(input);
const parsed = JSON.parse(fs.readFileSync(file, "utf8"));
if (!parsed.videoPath) throw new Error("videoPath is required");
const video = path.resolve(path.dirname(file), parsed.videoPath);
if (!fs.existsSync(video)) throw new Error(`Video not found: ${video}`);
if (!parsed.title) throw new Error("title is required");

console.log(
  JSON.stringify(
    {
      ok: true,
      title: parsed.title,
      videoPath: video,
      size_mb: (fs.statSync(video).size / 1024 / 1024).toFixed(2),
      tags: parsed.tags ?? [],
    },
    null,
    2,
  ),
);
